docs/migration-2.0.md

```diff
@@ -0,0 +1,1178 @@
+
+opts := []*options.ClientOptions{opts1, opts2}
+_, err := mongo.Connect(opts...)
+```
+
+```go
+// v2
+
+opts1 := options.Client().SetAppName("appName")
+opts2 := options.Client().SetConnectTimeout(math.MaxInt64)
+
+// Option objects are "Listers" in v2, objects that hold a list of setters
+opts := []options.Lister[options.ClientOptions]{opts1, opts2}
+_, err := mongo.Connect(opts...)
+```
+
+#### Creating Options from Builder
+
+Since a builder is just a slice of option setters, users can create options directly from a builder:
+
+```go
+// v1
+
+opt := &options.ClientOptions{}
+opt.ApplyURI(uri)
+
+return clientOptionAdder{option: opt}
+```
+
+```go
+// v2
+
+var opts options.ClientOptions
+for _, set := range options.Client().ApplyURI(uri).Opts {
+  _ = set(&opts)
+}
+
+return clientOptionAdder{option: &opts}
+```
+
+### FindOneOptions
+
+The following types are not valid for a `findOne` operation and have been removed:
+
+- `BatchSize`
+- `CursorType`
+- `MaxAwaitTime`
+- `NoCursorTimeout`
+
+### Merge\*Options
+
+All functions that merge options have been removed in favor of a generic solution. See [GORIVER-2696 ](https://jira.mongodb.org/browse/GORIVER-2696)for more information.
+
+### MaxTime
+
+Users should time out operations using either the client-level operation timeout defined by `ClientOptions.Timeout` or by setting a deadline on the context object passed to an operation. The following fields and methods have been removed:
+
+- `AggregateOptions.MaxTime` and `AggregateOptions.SetMaxTime`
+- `ClientOptions.SocketTimeout` and `ClientOptions.SetSocketTimeout`
+- `CountOptions.MaxTime` and `CountOptions.SetMaxTime`
+- `DistinctOptions.MaxTime` and `DistinctOptions.SetMaxTime`
+- `EstimatedDocumentCountOptions.MaxTime` and `EstimatedDocumentCountOptions.SetMaxTime`
+- `FindOptions.MaxTime` and `FindOptions.SetMaxTime`
+- `FindOneOptions.MaxTime` and `FindOneOptions.SetMaxTime`
+- `FindOneAndReplaceOptions.MaxTime` and `FindOneAndReplaceOptions.SetMaxTime`
+- `FindOneAndUpdateOptions.MaxTime` and `FindOneAndUpdateOptions.SetMaxTime`
+- `GridFSFindOptions.MaxTime` and `GridFSFindOptions.SetMaxTime`
+- `CreateIndexesOptions.MaxTime` and `CreateIndexesOptions.SetMaxTime`
+- `DropIndexesOptions.MaxTime` and `DropIndexesOptions.SetMaxTime`
+- `ListIndexesOptions.MaxTime` and `ListIndexesOptions.SetMaxTime`
+- `SessionOptions.DefaultMaxCommitTime` and `SessionOptions.SetDefaultMaxCommitTime`
+- `TransactionOptions.MaxCommitTime` and `TransactionOptions.SetMaxCommitTime`
+
+This example illustrates how to define an operation-level timeout using v2, without loss of generality.
+
+### SessionOptions
+
+`DefaultReadConcern`, `DefaultReadPreference`, and `DefaultWriteConcern` are all specific to transactions started by the session. Rather than maintain three fields on the Session struct, v2 has combined these options into `DefaultTransactionOptions` which specifies a `TransactionOptions` object.
```

Comment:
```suggestion
`DefaultReadConcern`, `DefaultReadPreference`, and `DefaultWriteConcern` are all specific to transactions started by the session. Rather than maintain three fields on the `Session` struct, v2 has combined these options into `DefaultTransactionOptions` which specifies a `TransactionOptions` object.
```